compaction/compaction.cc

```diff
@@ -1117,7 +1127,11 @@ class regular_compaction : public compaction {
         auto monitor = std::make_unique<compaction_write_monitor>(sst, _table_s, maximum_timestamp(), _sstable_level);
         sstable_writer_config cfg = make_sstable_writer_config(_type);
         cfg.monitor = monitor.get();
-        return compaction_writer{std::move(monitor), sst->get_writer(*_schema, partitions_per_sstable(), cfg, get_encoding_stats()), sst};
+        // adjust the partitions based on the droppable tombstone estimate if the garbage collected sstable writer is enabled
+        uint64_t estimated_partitions = enable_garbage_collected_sstable_writer()
+            ? std::max(1UL, uint64_t(ceil(partitions_per_sstable() * (1 - average_estimate_of_droppable_tombstone_ratio()))))
+            : partitions_per_sstable();
```

Comment:
I am worried this might cause a regression when lots of data cannot be dropped. so bloom filters can be very small to the number of partition that landed in the output sstables.

I am thinking that regular output sstables don't need an adjustment, as we already have an estimation of number of sstables that can fit in a single sstable. if output reduces number of sstables from 10 to 2.5, then we have good estimation for the first 2 sstables and only the last one will be off by 50%, but that can be later fixed with your work to downsize the filter if estimation went off. makes sense?

Reply:
s/estimation of number of sstables/estimation of number of partitions/

Reply:
Ack. I understand. The ratio is now applied only to the gc'ed sstable.